src/App.js

```diff
@@ -1,28 +1,37 @@
 import React from 'react';
 import Range from './components/Range';
-import './components/range.css';
+import Modal from './components/Modal';
+import FunctionalModal from './components/FunctionalModal';
+import CounterWithHooks from './components/CounterWithHooks';
+import Dropdown from './components/Dropdown';
+import './components/modal.css';
 import './App.css';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      rangeVal: 0
+      isOpen: false
     }
-    this.updateRange = this.updateRange.bind(this);
   }
   
-  updateRange(val) {
-    this.setState({
-      rangeVal: val
-    })
-  } 
-  
   render() {
-    const { rangeVal } = this.state;
     return (
-      <div>
-        <Range range={rangeVal} updateRange={this.updateRange}/>
+      <div className='App'>
+        <Range />
+        <button 
+          className='modal-btn' 
+          onClick={ (e) => this.setState({ isOpen: true })}>
+        Open Modal
+        </button>
+        <Modal 
+          isOpen={this.state.isOpen} 
```

Comment:
aici ai putea sa il scoti din this.state la inceput de render function, si sa il folosesti doar ca si isOpen instead of this.state.isOpen.
`const { isOpen } = this.state`